### DISCUSSION TITLE
New Architecture: `TurboModules` & `Fabric`

### DISCUSSION THREAD

**Discussion by divanc** [adaptyteam/AdaptySDK-React-Native#71] [Q&A]
Issue #70 raised a question of implementing new RN architecture features to `react-native-adapty`. I'm not currently eligible to answer how to do this. I will try to research these things and publish my ~adventures~ discoveries here

**Comment by giuliobracci** [comment]
Hi @divanc, now that the new architecture is the standard de-facto, are you planning on adopting it?

**Comment by akadko** [comment]
Agree, currently Adapty blocks main thread on each call (e.g. `.getPaywall()`)

**Comment by AdamGerthel** [comment]
Is this still not implemented? New arch is enabled by default in Expo now. I was just about to move from React Native IAP to Adapty and purchase a subscription but if new arch isn't supported, I'll need to use RevenueCat instead.

**Reply by StanislavMayorov** [reply]
Hi! Adapty RN SDK is compatible with new arch and tested with latest RN (new arch is enabled by default)

**Reply by AdamGerthel** [reply]
Ahhh - I wish the answer would've come sooner. I just switched to expo-iap. Good to know either way though. Might switch to Adapty to get receipt validation.

**Reply by StanislavMayorov** [reply]
We will add this information to the README. Also, this status has been added to the community repo: https://reactnative.directory/?search=adapty

Sorry to hear that! If you have any questions or need assistance with implementation, feel free to contact us at support@adapty.io - our support team responds quickly!
